aldryn_search/tests.py

```diff
@@ -96,9 +108,391 @@ def test_page_title_is_indexed_using_update_object(self):
 
         from cms.models import Title
         index = unified_index.get_index(Title)
-
         title = Title.objects.get(pk=page.title_set.all()[0].pk)
         index.update_object(title, using=DEFAULT_ALIAS)
         indexed = index.prepared_data
         self.assertEqual('home', indexed['title'])
         self.assertEqual('home', indexed['text'])
+
+class PluginFilterIndexingTests(TestCase):
+
+    def setUp(self):
+        self.request = get_request(language='en')
+
+    def test_page_title_is_indexed_using_prepare_with_filter_option(self):
+        """This tests the indexing path way used by update_index mgmt command"""
+        page = create_page(title="test_page", reverse_id='testpage', template="test.html", language="en")
+        plugin = add_plugin(page.placeholders.get(slot='content'), NotIndexedPlugin, 'en')
+        plugin2 = add_plugin(page.placeholders.get(slot='hidden_content'), HiddenPlugin, 'en')
+
+
+        from haystack import connections
```

Comment:
best to create a base test case and provide something like get_title_index() which can have the following:

```python
def get_title_index(self):
    search_conn = connections[DEFAULT_ALIAS]
    unified_index = search_conn.get_unified_index()
    index = unified_index.get_index(Title)
    return index
```